cubed/core/array.py

```diff
@@ -23,6 +26,10 @@ def gensym(name="array"):
     return f"{name}-{sym_counter:03}"
 
 
+# type representing either a CoreArray or a public-facing Array
+ChunkedArray = TypeVar("ChunkedArray", bound="CoreArray")
```

Comment:
Maybe name this `T_ChunkedArray` or similar? I been vaguely thinking it might be better to call `CoreArray` `ChunkedArray`, or it might be useful for something else in the future.